### PR TITLE
User Profile Terms of service

### PR DESCRIPTION
**Pull request by brianteeman**
In the user profile plugin there is a Terms of Service field that is required on registration. It has an optional accompanying field to link to an article.

Pull Request for Issue #28537 .

### Summary of Changes
Before this PR the logic for checking that a user had agreed to the terms of service was broken so thta if there is no linked article the cjeck is never performed and a user can still register even if they dont agree.

This PR removes the check for the existence of the article and tweaks the language string used in the error message accordingly

### Testing Instructions
- enable user registration
- enable the user profile plugin
- try to create a new user without agreeing to the terms

### Actual result BEFORE applying this Pull Request
User created

### Expected result AFTER applying this Pull Request
User not created and error message displayed

### CODE REVIEW

**Comment by Kostelano** [comment]
I have tested this item :white_check_mark: successfully on 420ed1e38970d7849189161abf9f1f1a73cdd79d<br /><br />Thanks, it works.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38619">issues.joomla.org/tracker/joomla-cms/38619</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 420ed1e38970d7849189161abf9f1f1a73cdd79d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38619">issues.joomla.org/tracker/joomla-cms/38619</a>.</sub>

**Comment by richard67** [comment]
RTC <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38619">issues.joomla.org/tracker/joomla-cms/38619</a>.</sub>

File: plugins/user/profile/profile.php [hunk 1]
```diff
@@ -332,7 +332,7 @@ public function onUserBeforeSave($user, $isnew, $data)
         $tosEnabled = ($this->params->get('register-require_tos', 0) == 2);
 
         // Check that the tos is checked.
-        if ($task === 'register' && $tosEnabled && $tosArticle && $option === 'com_users' && !$data['profile']['tos']) {
```
Comment (Quy) [inline plugins/user/profile/profile.php:339]: Remove line 331.

**Comment by richard67** [comment]
@Quy Does it need new tests? From my point of view RTC is still ok, I've just reviewed that last change.

**Comment by Quy** [comment]
It is not necessary as it is an obsolete variable.

**Comment by roland-d** [comment]
Thank you

**Comment by brianteeman** [comment]
thanks

**Pull request merged**